nsxt/resource_nsxt_policy_role.go

```diff
@@ -0,0 +1,266 @@
+					Schema: map[string]*schema.Schema{
+						"feature": {
+							Type:        schema.TypeString,
+							Description: "Feature Id",
+							Required:    true,
+						},
+						"feature_description": {
+							Type:        schema.TypeString,
+							Description: "Feature Description",
+							Computed:    true,
+						},
+						"feature_name": {
+							Type:        schema.TypeString,
+							Description: "Feature Name",
+							Computed:    true,
+						},
+						"permission": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(featurePermissionTypes, false),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceNsxtPolicyUserManagementRoleExists(id string, connector client.Connector) (bool, error) {
+	var err error
+	roleClient := aaa.NewRolesClient(connector)
+	_, err = roleClient.Get(id)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
+}
+
+func getFeaturePermissionFromSchema(d *schema.ResourceData) []nsxModel.FeaturePermission {
+	features := d.Get("feature").([]interface{})
+	featurePermissions := make([]nsxModel.FeaturePermission, 0, len(features))
+	for _, feature := range features {
+		data := feature.(map[string]interface{})
+		featureID := data["feature"].(string)
+		featureDescription := data["feature_description"].(string)
+		featureName := data["feature_name"].(string)
+		permission := data["permission"].(string)
+		featurePermissions = append(featurePermissions, nsxModel.FeaturePermission{
+			Feature:            &featureID,
+			FeatureDescription: &featureDescription,
+			FeatureName:        &featureName,
+			Permission:         &permission,
+		})
+	}
+
+	return featurePermissions
+}
+
+func setFeaturePermissionInSchema(d *schema.ResourceData, permissions []nsxModel.FeaturePermission) {
+	var permissionList []map[string]interface{}
+	for _, permission := range permissions {
+		// NSX returns the complete feature set on GET
+		// Bypass ineffective items from the list
+		if permission.IsInternal != nil && *permission.IsInternal {
+			// Internal permissions is not effective for roles
+			continue
+		}
+		if permission.Permission != nil && *permission.Permission == nsxModel.FeaturePermission_PERMISSION_NONE {
+			// Permissions are none by default. None-valued permissions are not allowed either in schema.
+			continue
+		}
+
+		elem := make(map[string]interface{})
+		elem["feature"] = permission.Feature
+		elem["feature_description"] = permission.FeatureDescription
+		elem["feature_name"] = permission.FeatureName
+		elem["permission"] = permission.Permission
+		permissionList = append(permissionList, elem)
+	}
+	err := d.Set("feature", permissionList)
+	if err != nil {
+		log.Printf("[WARNING] Failed to set feature in schema: %v", err)
+	}
+}
+
+func policyUserManagementRolePatch(id string, d *schema.ResourceData, m interface{}) error {
```

Comment:
Since the actual API call is `Update` and not `Patch`, I would suggest to change function name to improve readability

Reply:
Renamed.